extension/datadogfleetautomationextension/transform.go

```diff
@@ -0,0 +1,353 @@
+	for _, field := range []struct {
+		kind string
+		data map[component.Type]service.ModuleInfo
+	}{
+		{receiverKind, e.moduleInfo.Receiver},
+		{processorKind, e.moduleInfo.Processor},
+		{exporterKind, e.moduleInfo.Exporter},
+		{extensionKind, e.moduleInfo.Extension},
+		{connectorKind, e.moduleInfo.Connector},
+		// TODO: add Providers and Converters after upstream change accepted to add these to moduleinfos
+	} {
+		if componentKind == field.kind {
+			if _, ok := field.data[component.MustNewType(componentType)]; ok {
+				return true
+			}
+		}
+	}
+	return false
+}
+
+// isHealthCheckV2Enabled checks if healthcheckv2 is properly configured
+// should be ran after e.isComponentConfigured for healthcheckv2 and e.getComponentConfig for healthcheckv2
+func (e *fleetAutomationExtension) isHealthCheckV2Enabled() (bool, error) {
+	if useV2, ok := e.healthCheckV2Config["use_v2"].(bool); ok && useV2 {
+		if httpConfig, ok := e.healthCheckV2Config["http"].(map[string]any); ok {
+			if statusConfig, ok := httpConfig["status"].(map[string]any); ok {
+				if enabled, ok := statusConfig["enabled"].(bool); ok && enabled {
+					return true, nil
+				}
+				return false, errors.New("healthcheckv2 extension is enabled but http status check is not enabled; component status will not be available")
+			}
+			return false, errors.New("healthcheckv2 extension is enabled but http status is not configured; component status will not be available")
+		}
+		return false, errors.New("healthcheckv2 extension is enabled but http endpoint is not configured; component status will not be available")
+	}
+	return false, errors.New("healthcheckv2 extension is enabled but is set to legacy mode; component status will not be available")
+}
+
+// getComponentHealthStatus is subject to change/removal as healthcheckv2 is in "development status"
+// unclear if we should wait until this is more stabilized to add individual component status health
+// to the payloads we send
+//
+// requires e.componentStatus to be recently updated with e.getHealthCheckStatus() prior to function call
+func (e *fleetAutomationExtension) getComponentHealthStatus(id string, componentsKind string) map[string]any {
+	componentKind, ok := kindsToKind[componentsKind]
+	if !ok {
+		return nil
+	}
+	result := map[string]any{}
+	// scrape components list for extensions, pipelines list for receivers, processors, exporters, connectors
+	if componentsConfig, ok := e.componentStatus["components"].(map[string]any); ok {
+		if componentsKind == extensionsKind {
+			if componentStatus, ok := componentsConfig[componentsKind].(map[string]any); ok {
+				if receiversStatus, ok := componentStatus["components"].(map[string]any); ok {
+					componentNameInPipeline := componentKind + ":" + id
+					if componentStatus, ok := receiversStatus[componentNameInPipeline].(map[string]any); ok {
+						result = componentStatus
+					}
+				}
+			}
```

Comment:
We should try to avoid arrow pattern. see https://blog.codinghorror.com/flattening-arrow-code/
try changing to 

```
if componentsKind != extensionsKind {
continue 
}
```